syntax = "proto3";
import "common.proto";

package uart;

enum startstop_enum {
  STOP = 0;  // Not Ok
  START = 1; // Ok
}

enum steering_enum {
    RIGHT = 0;
    LEFT = 1;

}

enum stateWorking_enum {
  ACTIVE = 0;
  INACTIVE = 1;
  ERROR = 2;
}

enum brake_enum {
  RELEASE = 0;
  LOCK = 1;
}

message StartStop_req { startstop_enum cmd = 1; }

message Steering_req {
  steering_enum dir = 1;
  double angle = 2;
}

message Throttle_req { uint32 throttleValue = 1; }

message Brake_req { brake_enum brakeValue = 1; }

message StateWorking { stateWorking_enum cmd = 1; }

message HCSR4 { double distance = 1; }

message GPS {
  float latitude = 1;
  float longitude = 2;
}

enum pub_sub_message {
  START_STOP_MSG = 0;
  THROTTLE_MSG = 1;
  STEERING_MSG = 2;
  BRAKE_MSG = 3;
  STATE_WORKING_MSG = 4;
  HCSR4_MSG = 5;
  GPS_MSG = 6;
}

message pub_sub {
  pub_sub_message msg_type = 1;
  oneof messages {
    StartStop_req startstop = 2;
    Throttle_req throttle = 3;
    Steering_req steering = 4;
    Brake_req brake = 5;
    StateWorking statework = 6;
    HCSR4 hcsr4_dis = 7;
    GPS location = 8;
  }
}